File: apiEZPZ/src/main/java/com/example/apiezpz/checklist/domain/Category.java
package com.example.apiezpz.checklist.domain;

import lombok.*;
import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter @Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString(exclude = {"checklist", "checklistItems"})
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "checklist_id")
    private Checklist checklist;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ChecklistItem> checklistItems = new ArrayList<>();

    // 아이템 추가 편의 메서드
    public void addItem(ChecklistItem checklistItem) {
        checklistItems.add(checklistItem);
        checklistItem.setCategory(this);
    }

    // 아이템 삭제 편의 메서드
    public void removeItem(ChecklistItem checklistItem) {
        checklistItems.remove(checklistItem);
        checklistItem.setCategory(null);
    }
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/checklist/domain/Category.kt
package com.example.apiezpz.checklist.domain

import jakarta.persistence.*

@Entity
@Table(name = "categories")
data class Category(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    
    @Column(nullable = false)
    var name: String = "",

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "checklist_id")
    var checklist: Checklist? = null,

    @OneToMany(mappedBy = "category", cascade = [CascadeType.ALL], orphanRemoval = true)
    var checklistItems: MutableList<ChecklistItem> = mutableListOf()
) {
    // 아이템 추가 편의 메서드
    fun addItem(checklistItem: ChecklistItem) {
        checklistItems.add(checklistItem)
        checklistItem.category = this
    }

    // 아이템 삭제 편의 메서드
    fun removeItem(checklistItem: ChecklistItem) {
        checklistItems.remove(checklistItem)
        checklistItem.category = null
    }
} 